investors/views.py

```diff
@@ -0,0 +1,82 @@
+from django.db import transaction
+from rest_framework import viewsets, status, permissions
+from rest_framework.response import Response
+
+from .models import Investor
+from .serializers import InvestorSerializer
+
+
+class IsInvestorPermission(permissions.BasePermission):
+    """
+    Custom permission to only allow investors to interact with the view.
+    """
+    def has_permission(self, request, view):
+        return request.user.is_authenticated
+
+
+class InvestorViewSet(viewsets.ViewSet):
+    def get_permissions(self):
+        if self.action == 'list' or self.action == 'retrieve':
+            return []
+        else:
+            return [IsInvestorPermission()]
+
+    def list(self, request):
+        investors = Investor.objects.all()
+        serializer = InvestorSerializer(investors, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+    def retrieve(self, request, pk=None):
```

Comment:
looks much better, but try to investigate the possibility of implementation.

Example:
from django.shortcuts import get_object_or_404


def my_view(request):
    obj = get_object_or_404(MyModel, pk=1)

https://docs.djangoproject.com/en/5.0/topics/http/shortcuts/

Reply:
in our code, such an implementation is already found in **line 97** of this file:
_https://github.com/Project-Stage-Academy/UA_1155_alpha/blob/main/users/views.py_

Reply:
I just created separate app for investors and moved all code which concerns investors to this folder. I didn`t change methods. Just checked, if they works. 
We have another task for get_object_or_404 and others fixes for investors